server/container_create.go

```diff
@@ -1038,14 +1038,25 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 	}
 
 	// Get imageName and imageRef that are later requested in container status
-	status, err := s.StorageImageServer().ImageStatus(s.ImageContext(), images[0])
-	if err != nil {
-		return nil, err
+	var (
+		imgResult    *storage.ImageResult
+		imgResultErr error
+	)
+	for _, img := range images {
+		status, err := s.StorageImageServer().ImageStatus(s.ImageContext(), img)
+		if err == nil {
+			imgResult = status
+			break
+		}
+		imgResultErr = err
+	}
+	if imgResult == nil {
+		return nil, imgResultErr
```

Comment:
Instead of checking `imgResult` here, I'd rather have:

```go
if imgResultErr != nil {
    return nil, imgResultErr
}
```

You may *also* want to follow that with:

```go
if imgResult == nil {
    return nil, errors.Error("ImageStatus() is really broken and should have returned an error")
}
```

and you may want to guard before the loop with:

```go
if len(images) == 0 {
  return nil, errors.Error("cannot create a sandbox when images is empty")
}
```

unless we have an earlier guard for that (or know it will never be empty).

Reply:
:+1 especially on the err check vs result check.

Reply:
fixed.